permutations_test.py

```diff
@@ -0,0 +1,14 @@
+from permutations import permutations
+
+
+def test_permutaions():
+    assert permutations("a") == ["a"]
+    assert all(perm in ["ab", "ba"] for perm in permutations("ab"))
+    assert all(
+        perm in ["abc", "acb", "cba", "cab", "bca", "bac"]
+        for perm in permutations("abc")
+    )
+    assert all(
+        perm in ["aabb", "abab", "abba", "baab", "baba", "bbaa"]
+        for perm in permutations("aabb")
+    )
```

Comment:
Since you are checking for inclusion here, make these sets.  `all(perm in {"ab", "ba"} for....`

What is the benefit of using a set?

Reply:
With sets it is not necessary to iterate through their entirety like lists, as they are stored in hash tables which makes the code more efficient